src/lumigo_tracer/utils.py

```diff
@@ -199,12 +209,8 @@ def report_json(region: Union[None, str], msgs: List[dict]) -> int:
     host = Configuration.host or EDGE_HOST.format(region=region)
     duration = 0
     if not edge_connection or edge_connection.host != host:
-        try:
-            edge_connection = http.client.HTTPSConnection(  # type: ignore
-                host.lstrip("https://").rstrip(EDGE_PATH), timeout=EDGE_TIMEOUT
-            )
-        except Exception as e:
-            get_logger().exception(f"Could not establish connection to {host}", exc_info=e)
+        edge_connection = establish_connection(host)
+        if not edge_connection:
```

Comment:
It looks like you had a bug here before your change, returning duration although you had an error, am I right?

Reply:
nope. the function that calls it expects to get duration, and duration=0 is what returned in case of an error